lib/kml/geometry.ts

```diff
@@ -99,10 +99,21 @@ export function getGeometry(node: Element): GeometriesAndTimes {
   const geometries: Geometry[] = [];
   const coordTimes = [];
   for (const t of ["MultiGeometry", "MultiTrack", "gx:MultiTrack"]) {
-    const elem = get1(node, t);
-    if (elem) {
-      return getGeometry(elem);
+    const n = node.getElementsByTagName(t);
```

Comment:
This is a bug: with the example testcase, you'll notice that each LineString is represented multiple times in the converted output. That's because, when we just select all MultiGeometry elements here, the LineString is pulled from the top-level MultiGeometry _as well as_ the one contained in it.

This will need to be stricter to avoid adding a bug. So something like checking that each element is a direct child, and making this method recursive.

Reply:
Made getGeometry recursive